File: csharp/Behavioral/14-Command/Command/CompleteJobCommand.cs
public class CompleteJobCommand : ICommand
{
    private readonly JobCard _jobCard;
    private string           _previousStatus = string.Empty;

    public CompleteJobCommand(JobCard jobCard) => _jobCard = jobCard;

    public void Execute()
    {
        _previousStatus = _jobCard.Status;
        _jobCard.Status = "Completed";
        Console.WriteLine($"  Completed {_jobCard.JobId}. Status → Completed");
    }

    public void Undo()
    {
        _jobCard.Status = _previousStatus;
        Console.WriteLine($"  Undo complete: {_jobCard.JobId} Status → {_previousStatus}");
    }
}

File: typescript/Behavioral/14-Command/Command/CompleteJobCommand.ts
import { ICommand } from "./ICommand";
import { JobCard } from "../Model/JobCard";

export class CompleteJobCommand implements ICommand {
  private previousStatus: string = "";

  constructor(private readonly jobCard: JobCard) {}

  execute(): void {
    this.previousStatus = this.jobCard.status;
    this.jobCard.status = "Completed";
    console.log(`  Completed ${this.jobCard.jobId}. Status → Completed`);
  }

  undo(): void {
    this.jobCard.status = this.previousStatus;
    console.log(
      `  Undo complete: ${this.jobCard.jobId} Status → ${this.previousStatus}`,
    );
  }
}
